### PR TITLE
Feature/admin update embargo [OSF-7846]

### PR DESCRIPTION
**Pull request by laurenrevere**
## Purpose

Allows OSF admins to change the embargo date of a registration. This allows our admins to make changes easily if they make a mistake in their registration.

## Changes

Added a button to the admin app with this functionality.
Adds info about how to set up the admin app to the readme.

## Side effects

None

## Ticket

https://openscience.atlassian.net/browse/OSF-7846

modal-admin-embargo

### CODE REVIEW

File: admin/nodes/views.py [hunk 1]
```diff
@@ -228,6 +233,34 @@ def get_context_data(self, **kwargs):
...
+            change_embargo_date(registration, user, end_date)
+        except ValidationError as e:
+            return HttpResponse(e, status=409)
+        except PermissionDenied as e:
+            return HttpResponse(e, status=403)
+
+        return redirect(reverse_node(self.kwargs.get('guid')))
+
+    def get_object(self, queryset=None):
+        return Registration.load(self.kwargs.get('guid'))
```
Comment (alexschiller) [inline admin/nodes/views.py]: `Registration.objects.get(_id=self.kwargs.get('guid'))`
`load` is old syntax. It still exists but I think all refs have been removed for the most part. (probably for us to be lazy when using shell)

File: admin_tests/base/test_utils.py [hunk 2]
```diff
@@ -108,3 +111,41 @@ def test_rules_to_subjects(self):
...
+class TestNodeChanges(AdminTestCase):
+    def setUp(self):
+        super(TestNodeChanges, self).setUp()
+        self.registration = RegistrationFactory()
+        self.user = UserFactory()
+        self.user.is_staff = True
+        self.user.groups.add(Group.objects.get(name='osf_admin'))
+        self.user.save()
+
+        self.date1 = self.registration.registered_date + datetime.timedelta(days=365)
```
Comment (alexschiller) [inline admin_tests/base/test_utils.py]: These could be more descriptive to clarify the expected outcome -- date_valid, date_long_invalid,  etc

File: admin_tests/base/test_utils.py [hunk 3]
```diff
@@ -108,3 +111,41 @@ def test_rules_to_subjects(self):
...
+        #Update an embargo end date
+        change_embargo_date(self.registration, self.user, self.date2)
+        assert_almost_equal(self.registration.embargo.end_date, self.date2, delta=datetime.timedelta(days=1))
+
+        #Test an invalid date
+        with assert_raises(ValidationError):
+            change_embargo_date(self.registration, self.user, self.date3)
+
+        # Test that checks user has permission
+        with assert_raises( PermissionDenied):
```
Comment (alexschiller) [inline admin_tests/base/test_utils.py]: extra space before PermissionDenied

File: admin_tests/base/test_utils.py [hunk 4]
```diff
@@ -108,3 +111,41 @@ def test_rules_to_subjects(self):
...
+        self.registration = RegistrationFactory()
+        self.user = UserFactory()
+        self.user.is_staff = True
+        self.user.groups.add(Group.objects.get(name='osf_admin'))
+        self.user.save()
+
+        self.date1 = self.registration.registered_date + datetime.timedelta(days=365)
+        self.date2 = self.registration.registered_date + datetime.timedelta(days=375)
+        self.date3 = self.registration.registered_date + datetime.timedelta(days=1825)
+        self.registration.registered_date
```
Comment (alexschiller) [inline admin_tests/base/test_utils.py]: extra

File: admin_tests/base/test_utils.py [hunk 5]
```diff
@@ -108,3 +111,41 @@ def test_rules_to_subjects(self):
...
+        super(TestNodeChanges, self).setUp()
+        self.registration = RegistrationFactory()
+        self.user = UserFactory()
+        self.user.is_staff = True
+        self.user.groups.add(Group.objects.get(name='osf_admin'))
+        self.user.save()
+
+        self.date1 = self.registration.registered_date + datetime.timedelta(days=365)
+        self.date2 = self.registration.registered_date + datetime.timedelta(days=375)
+        self.date3 = self.registration.registered_date + datetime.timedelta(days=1825)
```
Comment (alexschiller) [inline admin_tests/base/test_utils.py]: A `too soon` date might also be a good test.

File: admin/templates/nodes/node.html [hunk 6]
```diff
@@ -1,5 +1,8 @@
 {% extends 'base.html' %}
 {% load static %}
+{% block top_includes %}
+    <script src="/static/js/nodes/registrations.js"></script>
```
[the remark this answers is not in the document]
Comment (laurenrevere) [reply admin/templates/nodes/node.html]: @alexschiller It's new. I forgot to track it!

**Comment by brianjgeiger** [comment]
@laurenrevere Conflicts

**Pull request merged**